core/src/main/java/org/hisp/dhis/android/core/arch/db/querybuilders/internal/SQLStatementBuilderImpl.kt

```diff
@@ -193,7 +192,7 @@ internal class SQLStatementBuilderImpl internal constructor(
         // TODO refactor to only generate for object without uids store.
         val whereClause =
             if (whereColumns.isEmpty()) {
-                CoreColumns.ID + " = -1"
```

Comment:
The code should never enter this part of flow (check the comment above). This condition should always be false to avoid updating/deleting any property that has an empty "where" columns. The "0" has the same effect.